src/Umbraco.Web.BackOffice/Services/IconService.cs

```diff
@@ -174,33 +172,30 @@ private static IEnumerable<FileInfo> GetIconsFiles(IFileProvider fileProvider, s
             // Iterate through the sub directories of each plugin folder
```

Comment:
There's a lot of code her for iterating paths and subpaths. It works just as intended, but it feels like it could be done with less code - something like:

```
            foreach (IFileInfo fileInfo in fileProvider
                         .GetDirectoryContents($"{path}/{pluginDirectory.Name}/{Constants.SystemDirectories.PluginIcons}")
                         .Where(fileInfo => fileInfo.Name.InvariantEndsWith(".svg") && fileInfo.PhysicalPath.IsNullOrWhiteSpace() == false))
            {
                yield return new FileInfo(fileInfo.PhysicalPath!);
            }
```

...but then again, maybe all this code is here to support invariant paths? In that case we need all of it 👍

Reply:
Also note that `fileInfo.PhysicalPath` can be `null` if the file isn't located on a physical file system (e.g. embedded resource, Azure Blob Storage, etc.). I suggest adding a check and optionally log a warning in this PR and refactor this to return `IFileInfo` in a separate PR...

Reply:
Aye 😄 there is a lot of null-or-whitespace checking going on for `PhysicalPath` all over the affected controllers, hopefully we should be good for this PR.

Reply:
Kenn, those nested loops were there from a [previous PR](https://github.com/umbraco/Umbraco-CMS/pull/12833) and in the [commit ](https://github.com/umbraco/Umbraco-CMS/pull/12833/commits/6e3598d05ebead3ce489133782480383e538d975)from @nul800sebastiaan indicates that your suggestion might not work in case of Linux 🤷

Reply:
@elit0451 yeah I did notice they didn't originate from your PR 😄 still... could you add a comment while we have it on hand, so no-one comes around to "fix" it later (and thus break Linux again)?

Reply:
Sure 👍